docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -11,410 +11,657 @@ The `ERC20.cairo` contract implements an approximation of https://eips.ethereum.
-
-=== ERC20Mintable
+More presets coming soon!
 
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Mintable.cairo[`ERC20Mintable`] preset allows the contract owner to mint new tokens.
-
-=== ERC20Pausable
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Pausable.cairo[`ERC20Pausable`] preset allows the contract owner to pause/unpause all state-modifying methods i.e.
-`transfer`, `approve`, etc.
-This preset proves useful for scenarios such as preventing trades until the end of an evaluation period and having an emergency switch for freezing all token transfers in the event of a large bug.
+=== ERC20 (basic)
 
-=== ERC20Upgradeable
+*TODO: CHANGE LINK TO RELEASE COMMIT WHEN AVAILABLE*
 
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Upgradeable.cairo[`ERC20Upgradeable`] preset allows the contract owner to upgrade a contract by deploying a new ERC20 implementation contract while also maintaining the contract's state.
-This preset proves useful for scenarios such as eliminating bugs and adding new features.
-For more on upgradeability, see xref:proxies.adoc#contract_upgrades[Contract upgrades].
+The https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/token/erc20/erc20.cairo[`ERC20`] preset offers a quick and easy setup for deploying a basic ERC20 token.
 
 == API Specification
 
-=== Methods
-
-[,cairo]
+=== ERC20Impl
+
+[,rust]
+----
+fn name(self: @TState) -> felt252;
+fn symbol(self: @TState) -> felt252;
+fn decimals(self: @TState) -> u8;
+fn total_supply(self: @TState) -> u256;
+fn balance_of(self: @TState, account: ContractAddress) -> u256;
+fn allowance(
+    self: @TState,
+    owner: ContractAddress,
+    spender: ContractAddress
+) -> u256;
+fn transfer(
+    ref self: TState,
+    recipient: ContractAddress,
+    amount: u256
+) -> bool;
+fn transfer_from(
+    ref self: TState,
+    sender: ContractAddress,
+    recipient: ContractAddress,
+    amount: u256
+) -> bool;
+fn approve(
+    ref self: TState,
+    spender: ContractAddress,
+    amount: u256
+) -> bool;
 ----
-func name() -> (name: felt) {
-}
 
-func symbol() -> (symbol: felt) {
-}
-
-func decimals() -> (decimals: felt) {
-}
-
-func totalSupply() -> (totalSupply: Uint256) {
-}
-
-func balanceOf(account: felt) -> (balance: Uint256) {
-}
-
-func allowance(owner: felt, spender: felt) -> (remaining: Uint256) {
-}
+==== `name`
 
-func transfer(recipient: felt, amount: Uint256) -> (success: felt) {
-}
+[.contract-item]
+==== `[.contract-item-name]#++name++#++(self: @TState) → felt252++`
 
-func transferFrom(sender: felt, recipient: felt, amount: Uint256) -> (success: felt) {
-}
+Returns the name of the token.
 
-func approve(spender: felt, amount: Uint256) -> (success: felt) {
-}
-----
+===== Returns
```

Comment:
We can remove this section here, and in the rest of the methods to match [solidity docs](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20-totalSupply--), because we are not documenting a CLI or NRE as in Nile, but a contract.